### PR TITLE
Update to support Ecto 2.2

### PR DESCRIPTION
**Pull request by rbeene**
* update `drop_autogenerated_ids` to pattern match against the 3-element tuple now present (https://github.com/elixir-ecto/ecto/commit/543e97abe4473e3b21442136eaf50f024cc28e60)

* bump Ecto to 2.2 in `mix.exs`

Note: there is a test failure currently that I haven't traced down yet but I wanted to jump start things with this WIP PR.

### CODE REVIEW

**Comment by rbeene** [comment]
Current test failure due to these changes. 

test_failure

**Comment by rbeene** [comment]
Looks like a potential fix for the test failure can be found in #238 
That PR currently is pointing Ecto to 2.1.6 vs 2.2 (released 2 days ago).

**Comment by syamilmj** [comment]
Anything we can do to help with this?

**Comment by paulcsmith** [comment]
Could you please try this branch in your project and report back on if this works without breaking anything else? I'll try to review this on some of our apps this week as well. Thank you so much for your help on this!

**Comment by rbeene** [comment]
@paulcsmith - I actually ran into this specific issue in a sample project I was doing for a lightning talk. The above does resolve the issue when running the latest Ecto.

**Comment by syamilmj** [comment]
This worked on my projects (two of them).

The test that failed here (test/ex_machina/ecto_strategy_test.exs:120) is the only instance where `ExMachina.Publisher` schema is being used and honestly I'm not sure what it is trying to achieve.

File: lib/ex_machina/ecto.ex [hunk 1]
```diff
@@ -301,7 +301,7 @@ defmodule ExMachina.Ecto do
 
   defp drop_autogenerated_ids(map, struct) do
     case struct.__schema__(:autogenerate_id) do
-      {name, _type} -> Map.delete(map, name)
+      {name, _source, _type,} -> Map.delete(map, name)
```
Comment (MrCoffey) [inline lib/ex_machina/ecto.ex]: Think you have a leading comma here

Comment (rbeene) [reply lib/ex_machina/ecto.ex]: Indeed. Thanks for the review!

**Comment by paulcsmith** [comment]
Moving forward with https://github.com/thoughtbot/ex_machina/pull/238 since it is backwards compatible. Thanks so much for this though.

I'm going to try to get this out by end of week. In the meantime you can point to master and it should work with Ecto 2.2

**Pull request closed without merging**